lib/neovim-plugin.nix

```diff
@@ -99,21 +103,47 @@ with lib;
             example = settingsExample;
           };
         }
+        // optionalAttrs allowLazyLoad {
+          lazyLoad = nixvimOptions.mkLazyLoadOption {
+            inherit originalName cfg;
+            optionsForPlugin = true;
+            lazyLoad =
+              {
+                after = optionalString callSetup ''
+                  require('${luaName}')${setup}(${optionalString (cfg ? settings) (toLuaObject cfg.settings)})
+                '';
+              }
+              // (optionalAttrs (isColorscheme && colorscheme != null) { colorscheme = [ colorscheme ]; })
+              // lazyLoad;
+          };
+        }
         // extraOptions;
 
       config =
         let
-          cfg = config.${namespace}.${name};
           extraConfigNamespace = if isColorscheme then "extraConfigLuaPre" else "extraConfigLua";
+          lazyLoaded = cfg.lazyLoad.enable or false;
         in
         mkIf cfg.enable (mkMerge [
           {
             extraPlugins = (optional installPackage cfg.package) ++ extraPlugins;
             inherit extraPackages;
 
-            ${extraConfigNamespace} = optionalString callSetup ''
+            ${extraConfigNamespace} = optionalString (callSetup && !lazyLoaded) ''
               require('${luaName}')${setup}(${optionalString (cfg ? settings) (toLuaObject cfg.settings)})
             '';
+
+            plugins.lz-n.plugins = mkIf lazyLoaded [ cfg.lazyLoad ];
+            assertions = [
+              {
+                assertion = config.plugins.lz-n.enable || !lazyLoaded;
+                # TODO: replace with global option after adding backend
```

Comment:
Maybe instead of configuring a backend, we could just use whichever supported `plugins.*` is enabled?

With this approach, we might want to warn/assert that only _one_ such implementation is enabled?